lib/index.ts

```diff
@@ -50,12 +51,70 @@ function analyzeDynamically(
   });
 }
 
-async function analyzeStatically(
-  targetImage: string,
-  options?: any,
-): Promise<never> {
+async function analyzeStatically(targetImage: string, options?: any) {
   const staticScanningOptions = getStaticAnalysisOptions(options);
-  return analyzer.analyzeStatically(targetImage, staticScanningOptions);
+
+  // Relevant only if using a Docker runtime. Optional, but we may consider what to put here
+  // to present to the user in Snyk UI.
+  const runtime = undefined;
+  // Both the analysis and the manifest files are relevant if inspecting a Dockerfile.
+  // This is not the case for static scanning.
+  const dockerfileAnalysis = undefined;
+  const manifestFiles = [];
+
+  try {
+    const staticAnalysis = await analyzer.analyzeStatically(
+      targetImage,
+      staticScanningOptions,
+    );
+
+    const parsedAnalysisResult = parseAnalysisResults(
+      targetImage,
+      staticAnalysis,
+      dockerfileAnalysis,
+    );
+
+    const dependenciesTree = await buildTree(
+      targetImage,
+      parsedAnalysisResult.type,
+      parsedAnalysisResult.depInfosList,
+      parsedAnalysisResult.targetOS,
+    );
+
+    const analysis = {
+      package: dependenciesTree,
+      packageManager: parsedAnalysisResult.type,
+      imageId: parsedAnalysisResult.imageId,
+      binaries: parsedAnalysisResult.binaries,
+      imageLayers: parsedAnalysisResult.imageLayers,
+    };
+
+    return buildResponse(
+      runtime,
+      analysis,
+      dockerfileAnalysis,
+      manifestFiles,
+      staticScanningOptions,
+    );
+  } catch (error) {
+    const analysisError = tryGetAnalysisError(error, targetImage);
+    throw analysisError;
+  }
+}
+
+function tryGetAnalysisError(error, targetImage): Error {
+  if (typeof error === "string") {
```

Comment:
I would check if error is an instance of `Error` and return it if it does. otherwise, perform your functionality

Reply:
I can include this in a chore commit at the end!

Reply:
I actually don't feel comfortable changing this, it's not a very well-tested path. I looked up some solutions with `instanceof` but I don't understand it really well so for both reasons I'll opt to leave the code as it is!